java/statestore-lambda/src/main/java/sleeper/statestore/committer/lambda/StateStoreCommitterLambda.java

```diff
@@ -58,24 +59,47 @@ public class StateStoreCommitterLambda implements RequestHandler<SQSEvent, SQSBa
     private final StateStoreCommitRequestDeserialiser serDe = new StateStoreCommitRequestDeserialiser();
 
     public StateStoreCommitterLambda() {
-        this(connectToAws());
-    }
+        AmazonS3 s3Client = AmazonS3ClientBuilder.defaultClient();
+        AmazonDynamoDB dynamoDBClient = AmazonDynamoDBClientBuilder.defaultClient();
+        String s3Bucket = System.getenv(CONFIG_BUCKET.toEnvironmentVariable());
+
+        InstanceProperties instanceProperties = new InstanceProperties();
+        instanceProperties.loadFromS3(s3Client, s3Bucket);
+        Configuration hadoopConf = HadoopConfigurationProvider.getConfigurationForLambdas(instanceProperties);
 
-    public StateStoreCommitterLambda(StateStoreCommitter committer) {
-        this.committer = committer;
+        TablePropertiesProvider tablePropertiesProvider = new TablePropertiesProvider(instanceProperties, s3Client, dynamoDBClient);
+        StateStoreProvider stateStoreProvider = new StateStoreProvider(instanceProperties, s3Client, dynamoDBClient, hadoopConf);
+        this.committer = new StateStoreCommitter(
+                CompactionJobStatusStoreFactory.getStatusStore(dynamoDBClient, instanceProperties),
+                IngestJobStatusStoreFactory.getStatusStore(dynamoDBClient, instanceProperties),
+                stateStoreProvider.byTableId(tablePropertiesProvider),
+                key -> s3Client.getObjectAsString(instanceProperties.get(DATA_BUCKET), key),
+                Instant::now);
     }
 
     @Override
     public SQSBatchResponse handleRequest(SQSEvent event, Context context) {
         Instant startTime = Instant.now();
         LOGGER.info("Lambda started at {}", startTime);
         List<BatchItemFailure> batchItemFailures = new ArrayList<>();
-        for (SQSMessage message : event.getRecords()) {
+        List<SQSMessage> messages = event.getRecords();
+        for (int i = 0; i < messages.size(); i++) {
+            SQSMessage message = messages.get(i);
             LOGGER.info("Found message: {}", message.getBody());
             StateStoreCommitRequest request = serDe.fromJson(message.getBody());
             try {
-                committer.apply(request);
-            } catch (RuntimeException | StateStoreException e) {
+                DynamoDBUtils.retryOnThrottlingException(() -> {
```

Comment:
We should probably be explicit about the polling settings here. It might be best to remove the method signature where you don't set that.

Reply:
More than that, there could be up to 10 messages in an invocation of the lambda. Right now if there's a throttling exception and it tries for 9 minutes on the first message before it's successful, it will reset the amount of time it'll wait for while processing the next message.

We really need a limit on the amount of time we'll wait for throttling across the whole invocation.

Reply:
Thanks, i've created this follow up issue to be done before this issue https://github.com/gchq/sleeper/issues/3011